lib/tbot/services/application/proxy_config.go

```diff
@@ -0,0 +1,98 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package application
+
+import (
+	"net"
+	"net/url"
+
+	"github.com/gravitational/trace"
+	"gopkg.in/yaml.v3"
+
+	"github.com/gravitational/teleport/lib/tbot/bot"
+	"github.com/gravitational/teleport/lib/tbot/internal/encoding"
+)
+
+// ProxyServiceType is the type string for the ProxyService, used within the
+// service header to indicate the service type.
+const ProxyServiceType = "application-proxy"
+
+// ProxyServiceConfig is the configuration for the ProxyService.
+type ProxyServiceConfig struct {
+	// Name of the service for logs and the /readyz endpoint.
+	// Optional.
+	Name string `yaml:"name,omitempty"`
+	// Listen is the address on which application proxy should listen. Example:
+	// - "tcp://127.0.0.1:3306"
+	// - "tcp://0.0.0.0:3306
```

Comment:
Extreme nitpick territory, but maybe we should recommend a port that doesn't conflict with a known non-http service 🤔